src/utils/ahash.zig

```diff
@@ -0,0 +1,209 @@
+const std = @import("std");
+const sig = @import("../sig.zig");
+
+const AtomicU64 = std.atomic.Value(u64);
+const Random = std.rand.Random;
+
+/// AHasher is a minimal impelementation of the AHash, a fast, DOS-resistant hash algorithm (https://github.com/tkaitchuck/aHash).
+/// Currently, only the functionality required to replicate the agave Deduper is implemented.
+pub const AHasher = struct {
+    buffer: u64,
+    pad: u64,
+    extra_keys: [2]u64,
+
+    // This constant comes from Kunth's prng (Empirically it works better than those from splitmix32).
+    const MULTIPLE: u64 = 6364136223846793005;
+    const ROTATE: u32 = 23;
+
+    pub fn fromRandomState(rand_state: RandomState) AHasher {
+        return .{
+            .buffer = rand_state.k0,
+            .pad = rand_state.k1,
+            .extra_keys = .{ rand_state.k2, rand_state.k3 },
+        };
+    }
+
+    pub fn hash(self: *AHasher, comptime T: type, data: *const T) void {
```

Comment:
This function should be removed, `write` should be made `pub` and renamed to `update`, and the current `update` should be renamed to `updateInt`.
This will allow a user to use `std.hash.autoHash[Strat]` for more complex data.

Reply:
This changes the behaviour and breaks dependant unit tests. I'd rather discuss and implement a solution in a follow up PR if need be